File: Data Structures & Algorithms/sqrtx/submission-0.cs
public class Solution {
    public int MySqrt(int x) {
        int left = 1; 
        int right = x;

        while(left<=right){
            int middle = left + (right - left) / 2;
            long middleSquere = (long)middle*middle;

            if(middleSquere == x){
                return middle;
            }else if (middleSquere < x){
                left = middle + 1;
            }else{
                right = middle - 1;
            }
   
        }

        return right;

    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.java
class Solution {
    public String longestPalindrome(String s) {
        int start = 0;
        int maxLen = 0;

        for(int i = 0; i < s.length(); i++){
            int currMax = Math.max(maxLength(s, i, i), maxLength(s, i, i+1));
            if(currMax > maxLen){
                maxLen = currMax;
                start = i - (currMax - 1) / 2;
            }
        }

        return s.substring(start, start + maxLen);

    }

    public int maxLength(String s, int left, int right){
        while(left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)){
            left--;
            right++;
        }

        return (right - 1) - left;
    }
}